packages/kbn-cell-actions/src/types.ts

```diff
@@ -20,6 +20,7 @@ export interface CellActionsProviderProps {
   getTriggerCompatibleActions: UiActionsService['getTriggerCompatibleActions'];
 }
 
+export type CellActionValue = string | string[] | null | undefined;
```

Comment:
This can be a problem, all cell actions expect this type currently. In Security turn them into strings, and everything works, are you okay with that? Otherwise, we'd have to refactor all the cell actions registered.

Reply:
Sorry, I am not familiar with how/why cell action logic is tied to string type only. Could you please link to that part of code?

There is this type casting in the PR which looks problematic as values coming from users data can be not only strings.
```
return displayedRows[pageRowIndex].flattened[fieldName] as CellActionValue;
```

Reply:
All the actions in Security:
https://github.com/elastic/kibana/tree/bb495c2d5088456208ca26980d7c4c122f2743d9/x-pack/plugins/security_solution/public/actions

and the generic ones in the package as well:
https://github.com/elastic/kibana/tree/bb495c2d5088456208ca26980d7c4c122f2743d9/packages/kbn-cell-actions/src/actions

expect the `CellActionValue` type in the value parameter for the `isCompatible` and `execute` functions, if we add the number and object types to it we would have to adapt all those actions. That would be way out of the scope of this task.

We can either block this PR until we add support for those types to the `CellActionValue` (not sure it is going to happen anytime soon) or we can try to find a workaround:

In Security we convert numbers to strings and everything works, it should not be a problem. But we do not have objects, we could try to stringify, but I think it would be better to just hide the actions for object values somehow. and use the default discover cell actions, that support objects.

What do you think?

Reply:
> In Security we convert numbers to strings and everything works, it should not be a problem. But we do not have objects, we could try to stringify, but I think it would be better to just hide the actions for object values somehow. and use the default discover cell actions, that support objects.

Sounds good to me if it fits Security context.